### PR TITLE
Store: allow configuration of series, series sample, and downloaded bytes limits

### PR DESCRIPTION
**Pull request by douglascamata**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

* Allow the configuration of the Store's touched series, series sample, and downloaded bytes limits.

### CODE REVIEW

**Comment by douglascamata** [comment]
Depends on #293.

**Review by yeya24 — APPROVED** [review]
Nice!

**Review by PhilipGough — APPROVED** [review]

File: jsonnet/kube-thanos/kube-thanos-store.libsonnet [hunk 1]
```diff
@@ -22,6 +22,9 @@ function(params) {
 
   // Safety checks for combined config of defaults and params
   assert std.isNumber(ts.config.replicas) && ts.config.replicas >= 0 : 'thanos store replicas has to be number >= 0',
+  assert std.isNumber(ts.config.limits.seriesTouched) && ts.config.limits.seriesTouched >= 0 : 'thanos store series touched limit has to be number >= 0',
```
Comment (saswatamcode) [inline jsonnet/kube-thanos/kube-thanos-store.libsonnet:25]: Let's maybe check if this is object as well?
```suggestion
  assert std.isObject(ts.config.limits) : 'thanos store series limit has be an object',
  assert std.isNumber(ts.config.limits.seriesTouched) && ts.config.limits.seriesTouched >= 0 : 'thanos store series touched limit has to be number >= 0',
```

**Review by saswatamcode — APPROVED** [review]
LGTM! Thanks!

Just one small suggestion.

**Review by matej-g — APPROVED** [review]

File: jsonnet/kube-thanos/kube-thanos-store.libsonnet (hunk 1 above)
Comment (douglascamata) [reply jsonnet/kube-thanos/kube-thanos-store.libsonnet:25]: Good idea, thanks! Will add this.

Comment (douglascamata) [reply jsonnet/kube-thanos/kube-thanos-store.libsonnet:26]: Done in https://github.com/thanos-io/kube-thanos/pull/292/commits/adf2f1c480eae000e83da97a31850c2580595c2e.

**Pull request merged**